Packages/com.unity.inputsystem/InputSystem/Plugins/Android/AndroidSupport.cs

```diff
@@ -39,7 +46,68 @@ public static void Initialize()
+            InputSystem.RegisterTemplate<AndroidOrientation>();
+            InputSystem.RegisterTemplate<AndroidGyroscope>();
+            InputSystem.RegisterTemplate<AndroidLight>();
+            InputSystem.RegisterTemplate<AndroidPressure>();
+            InputSystem.RegisterTemplate<AndroidProximity>();
+            InputSystem.RegisterTemplate<AndroidTemperature>();
+            InputSystem.RegisterTemplate<AndroidGravity>();
+            InputSystem.RegisterTemplate<AndroidLinearAcceleration>();
+            InputSystem.RegisterTemplate<AndroidRotationVector>();
+            InputSystem.RegisterTemplate<AndroidRelativeHumidity>();
+            InputSystem.RegisterTemplate<AndroidAmbientTemperature>();
+            InputSystem.RegisterTemplate<AndroidMagneticFieldUncalibrated>();
+            InputSystem.RegisterTemplate<AndroidGameRotationVector>();
+            InputSystem.RegisterTemplate<AndroidGyroscopeUncalibrated>();
+            InputSystem.RegisterTemplate<AndroidSignificantMotion>();
+            InputSystem.RegisterTemplate<AndroidStepDetector>();
+            InputSystem.RegisterTemplate<AndroidStepCounter>();
+            InputSystem.RegisterTemplate<AndroidGeomagneticRotationVector>();
+            InputSystem.RegisterTemplate<AndroidHeartRate>();
+
+            InputSystem.onFindTemplateForDevice += OnFindTemplateForDevice;
+        }
+
+        internal static string OnFindTemplateForDevice(int deviceId, ref InputDeviceDescription description,
+            string matchedTemplate, IInputRuntime runtime)
+        {
+            if (description.interfaceName != "Android" || string.IsNullOrEmpty(description.capabilities))
+                return null;
+
+            switch (description.deviceClass)
+            {
+                case "AndroidGameController":
+                {
+                    var caps = AndroidDeviceCapabilities.FromJson(description.capabilities);
+                    if ((caps.inputSources & AndroidInputSource.Gamepad) == AndroidInputSource.Gamepad)
+                    {
+                        if (caps.motionAxes != null)
+                        {
+                            if (caps.motionAxes.Contains(AndroidAxis.HatX) &&
+                                caps.motionAxes.Contains(AndroidAxis.HatY))
+                                return "AndroidGamepadWithDpadAxes";
+                        }
+                        return "AndroidGamepadWithDpadButtons";
+                    }
+
+                    return "AndroidJoystick";
+                }
+                case "AndroidSensor":
+                {
+                    var caps = AndroidSensorCapabilities.FromJson(description.capabilities);
+                    if (Enum.IsDefined(typeof(AndroidSenorType), caps.sensorType))
+                        return "Android" + caps.sensorType.ToString();
+                    return null;
+                }
+                default:
+                    return null;
+            }
```

Comment:
Please add tests.

Reply:
You can probably bulk-test all the added devices similar to `FunctionalTests.Devices_CanCreateDevice`.